addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -49,6 +50,20 @@ def _get_specific_rendering_values(self, processing_values):
         converted_amount = payment_utils.to_minor_currency_units(self.amount, self.currency_id)
         base_url = self.provider_id.get_base_url()
         return_url_params = {'reference': self.reference}
+
+        phone = self.partner_phone
+        if not self.partner_phone:
+            raise ValidationError("Razorpay: " + _("Phone number is required."))
+        else self.partner_phone:
```

Comment:
```suggestion
        else:
```
`else` comes at the end of all condition ;)